community/io/src/test/java/org/neo4j/graphdb/mockfs/EphemeralFileSystemAbstraction.java

```diff
@@ -345,7 +345,7 @@ public boolean renameFile( File from, File to ) throws IOException
     @Override
     public File[] listFiles( File directory )
     {
-        if ( files.containsKey( directory ) )
+        if ( files.containsKey( directory ) || !directories.contains( directory ) )
```

Comment:
Turns out it wasn't that easy. Changes causes `TestEphemeralFileChannel` to break. Looks like a more extensive refactoring of how EFSA works is needed to simulate the behaviour of `DefaultFileSystemAbstraction`. I will revisit this in 3.1 and close this PR.